### PR TITLE
add LogMonitor to check log disk freeBytes and change log level when space is almost full

### PR DESCRIPTION
**Pull request by Nivras**
#### What type of PR is this?
- [ ] bug
- [x] feature
- [ ] enhancement

#### What does this PR do?
add LogMonitor to check log disk freeBytes and change log level when space is almost full

in default config, if log disk free space is less than 256M, change log level to WARNING, and if space less than 64M, change log level to DEBUG, and less than 4M, change log level to FATAL.

If log disk space is bigger than those flag, and log level have changed before(maybe someone clean the log), LogMonitor will change the log level to the old value(which in config file).

#### Which issue(s)/PR(s) this PR relates to?
https://github.com/vesoft-inc/nebula/issues/3423
  
#### Checklist:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the corresponding label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes:

Please confirm whether to be reflected in release notes and how to describe:
>                                                                 `

### CODE REVIEW

File: src/storage/LogMonitor.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    freeByte_ = info.available;
+  } else {
+    LOG(WARNING) << "Get filesystem info of logdir: " << FLAGS_log_dir << " failed";
+  }
+}
+
+void LogMonitor::checkAndChangeLogLevel() {
+  getLogDiskFreeByte();
+
+  if (freeByte_ < FLAGS_log_min_reserved_bytes_to_warn) {
```
Comment (critical27) [inline src/storage/LogMonitor.cpp:34]: If log_level changed, maybe write some log to stderr or LOG.

**Review by critical27 — COMMENTED** [review]
Good job, generally LGTM, maybe you could move to `common` directory.

**Review by critical27 — APPROVED** [review]
Well done~~

**Comment by critical27** [comment]
Close #3423

File: src/common/log/LogMonitor.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    freeByte_ = info.available;
+  } else {
+    LOG(WARNING) << "Get filesystem info of logdir: " << FLAGS_log_dir << " failed";
+  }
+}
+
+void LogMonitor::checkAndChangeLogLevel() {
+  getLogDiskFreeByte();
+
+  if (freeByte_ < FLAGS_log_min_reserved_bytes_to_warn) {
```
Comment (pengweisong) [inline src/common/log/LogMonitor.cpp:33]: should first test the minimum size?

**Review by liuyu85cn — DISMISSED** [review]

File: src/common/log/LogMonitor.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+#include "common/log/LogMonitor.h"
+
+namespace nebula {
+
+DEFINE_uint64(log_min_reserved_bytes_to_warn,
+              256 * (1UL << 20),
```
Comment (liuyu85cn) [inline src/common/log/LogMonitor.cpp:10]: how about add some more readable comments to these 256/64/4?

**Review by critical27 — APPROVED** [review]

File: src/common/log/LogMonitor.cpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    freeByte_ = info.available;
+  } else {
+    LOG(WARNING) << "Get filesystem info of logdir: " << FLAGS_log_dir << " failed";
+  }
+}
+
+void LogMonitor::checkAndChangeLogLevel() {
+  getLogDiskFreeByte();
+
+  if (FLAGS_log_min_reserved_bytes_to_fatal > FLAGS_log_min_reserved_bytes_to_error ||
```
Comment (pengweisong) [inline src/common/log/LogMonitor.cpp:40]: I think `!(FLAGS_log_min_reserved_bytes_to_fatal > FLAGS_log_min_reserved_bytes_to_warn && FLAGS_log_min_reserved_bytes_to_error > FLAGS_log_min_reserved_bytes_to_fatal)` is more readable ?

**Review by pengweisong — APPROVED** [review]

**Pull request merged**